rules/terraformrules/terraform_empty_list_equality_test.go

```diff
@@ -41,6 +41,34 @@ resource "aws_db_instance" "mysql" {
 				},
 			},
 		},
+		{
+			Name: "comparing with [] is not recommended (mixed with other conditions)",
+			Content: `
+resource "aws_db_instance" "mysql" {
+	count = true == true || false != true && (false == false || [] == []) ? 1 : 0
```

Comment:
This test is unclear and seems to be trying to do to much. The goal here should be to have unit tests where the case being tested is clear and unique. That likely means more tests to achieve the same coverage, but the ability for a reader to understand the test's intent is paramount.

Reply:
separated this test in two (one to check parentheses and another one to check multiple conditions that check for empty lists)